core/services/relay/evm/evm.go

```diff
@@ -100,7 +101,7 @@ func NewRelayer(lggr logger.Logger, chain evm.Chain, opts RelayerOpts) (*Relayer
 		chain:            chain,
 		lggr:             lggr,
 		ks:               opts.CSAETHKeystore,
-		mercuryPool:      wsrpc.NewPool(lggr),
+		mercuryPool:      opts.MercuryPool,
```

Comment:
Why is this a top level pool now? Is the intention to share it across chains? That mans it will cross the relay API, and I'm not sure how that is meant to work :thinking:

Reply:
Yes it is shared across chains.

Reply:
So the relay API has to proxy the shared websocket connections? Why is that better than each plugin process having it's own local pool?

Reply:
I'm observing that this at least adds to the technical debt in the way of LOOPPifying mercury. 
1. If we intend to share these pools between host and plugins, then I guess that is just something to figure out how to do (but it does not immediately make sense to me).
2. If we do not intend to share these pools between host and plugins, then there is no need to refactor things to be global like this in the first place since we'll just have to undo it in a few weeks. Instead, each relayer should simply own and manage a pool internaally.

Reply:
Yes, good point.

One interesting thought that occurred to me is that this is a symptom of there not being a single "Mercury relayer", i.e. a symptom of Mercury being reimplemented across different LOOPP instances.

If we instead gave Mercury its own relayer, then all calls to Mercury would go via the same relayer and therefore would benefit from the wsrpc loopp which in that case would be a totally internal, but shared resource. This Mercury relayer would only implement the write APIs (i.e. the transmitter, the codec etc), and would leave the read side unimplemented or stubbed out.

The benefit of this is that it cleanly maps the relayer as a 1-1 abstraction over the underlying resource, in the same way that relayers do with the chains they target. As a result, you could instantiate the pool internally to the relayer and things would just naturally benefit from the pool.

It does mean that we'd need to refine the relayer abstraction somewhat, to allow products to use a different relayer for reading and for writing, but that doesn't feel wrong to me, and might even be a better underlying fit (CCIP might benefit from this for example).

Reply:
Yeah I think that separate reading and writing relayers makes sense :+1:

Reply:
@samsondav Do you think that is a good long-term direction for Mercury?

Reply:
I agree, separating the read/write relayers makes sense to me